src/backend/app/db/postgis_utils.py

```diff
@@ -90,10 +90,15 @@ def get_centroid(
 
 
 def geojson_to_geometry(
-    geojson: Union[FeatCol, Feature, Polygon],
+    geojson: Union[dict, FeatCol, Feature, Polygon],
 ) -> Optional[WKBElement]:
     """Convert GeoJSON to SQLAlchemy geometry."""
-    parsed_geojson = parse_and_filter_geojson(geojson.model_dump_json(), filter=False)
+    if isinstance(geojson, dict):
```

Comment:
i just noticed that we are using Feature and Polygon also from geojson_pydantic
```python 
from geojson_pydantic import Feature, Polygon
````

Reply:
The function you wrote `multipolygon_to_polygon` should probably be using Polygon from `geojson.Polygon` 👍

Reply:
i mean in geojson_to_geometry, they are from geojson_pydantic, so they probably need `model_dump_json` . And yes i understand your point, that the function i created uses geojson.Polygon, so it will pass Polygon to  geojson_to_geometry from geojson but not from pydantic.